### PR TITLE
Added utility functions to check schema types.

### PR DESCRIPTION
**Pull request by iwoplaza**

### CODE REVIEW

File: packages/typegpu/src/data/struct.ts [hunk 1]
```diff
@@ -121,13 +127,16 @@ class TgpuStructImpl<TProps extends Record<string, AnyTgpuData>>
...
+  if (isAlignedSchema(field as unknown)) {
     return `@align(${field.byteAlignment}) `;
   }
-  if (field instanceof TgpuSizedImpl) {
+
+  if (isSizedSchema(field as unknown)) {
     return `@size(${field.size}) `;
   }
+
+  return undefined;
```
Comment (mhawryluk) [inline packages/typegpu/src/data/struct.ts:139]: this line is unnecessary, was it added to make the undefined return more explicit?

File: packages/typegpu/src/data/struct.ts [hunk 2]
```diff
@@ -121,13 +127,16 @@ class TgpuStructImpl<TProps extends Record<string, AnyTgpuData>>
   }
 }
 
-function getAttribute(field: AnyTgpuData): string | undefined {
-  if (field instanceof TgpuAlignedImpl) {
+function getAttribute<T extends AnyTgpuData>(field: T): string | undefined {
```
Comment (mhawryluk) [inline packages/typegpu/src/data/struct.ts:130]: as noticed by @reczkok, this function doesn't work properly when both properties are present. we should fix that in another PR

File: packages/typegpu/src/data/struct.ts [hunk 3]
```diff
@@ -121,13 +127,16 @@ class TgpuStructImpl<TProps extends Record<string, AnyTgpuData>>
...
+  if (isAlignedSchema(field as unknown)) {
     return `@align(${field.byteAlignment}) `;
   }
-  if (field instanceof TgpuSizedImpl) {
+
+  if (isSizedSchema(field as unknown)) {
```
Comment (mhawryluk) [inline packages/typegpu/src/data/struct.ts:135]: why are these `as unknown` necessary?

**Review by mhawryluk — APPROVED** [review]
Great! 😊

File: packages/typegpu/src/data/struct.ts (hunk 1 above)
Comment (iwoplaza) [reply packages/typegpu/src/data/struct.ts:139]: Yess 🙌

File: packages/typegpu/src/data/struct.ts (hunk 3 above)
Comment (iwoplaza) [reply packages/typegpu/src/data/struct.ts:135]: For some reason, TypeScript assumes after each of these `if` statements, that the `field` property can be nothing else that that condition, so it becomes `never`. I will debug it and fix it in another PR when I figure out what is wrong.

**Comment by iwoplaza** [comment]
@reczkok Any thoughts before I merge? 👀

**Review by reczkok — APPROVED** [review]
Looks great💜

**Pull request merged**